BUILD.bazel

```diff
@@ -0,0 +1,94 @@
+# Copyright (c) 2021 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# DISCLAIMER: Bazel support is community-based. The maintainers do not
+# use Bazel internally. The Bazel build can have security risks or
+# optimization gaps.
+
+package(
+    default_visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "tbb12",
```

Comment:
`tbb12` is applicable only for Windows since it doesn't support convenient library versioning. While for Linux and macOS library name is still `tbb` but with soversioning and symlinks (`libtbb.so --> libtbb.so.12 --> libtbb.so.12.2`)

In CMake version/soversion are defined through target properties: https://github.com/oneapi-src/oneTBB/blob/ac3d43b691b69f943fee95668aa373c13d0d0d47/src/tbb/CMakeLists.txt#L86-L87

And then CMake does all the magic with soname and symlinks.

Is there any support for this in Bazel?

And maybe more general question: is Bazel about redistribution of binary packages or it is mostly about build from source each time?

Reply:
I renamed it from` tbb12` to `tbb` again. I am currently, not aware of a good way to handle soversion. The preferred Bazel way is to build and link all libraries static. I am currently not aware of a good strategy to support also dynamic linking using Bazel.

The preferred way of a Bazel build is to build everything from source each time (since Bazel has one of the best caching mechanisms rebuild times shrink dramatically)

Reply:
I'm ok with it, in case of static libraries you should not care about versioning. But if you want to use dynamic library you will have to care to ensure proper coexistence with other versions of this dynamic library (e.g. if it is pre-installed on the system).